src/idl_gen_cpp.cpp

```diff
@@ -2745,34 +2765,44 @@ class CppGenerator : public BaseGenerator {
 
     // Generate a default constructor.
     code_ += "  {{STRUCT_NAME}}() {";
-    code_ += "    memset(static_cast<void *>(this), 0, sizeof({{STRUCT_NAME}}));";
+    code_ +=
+        "    memset(static_cast<void *>(this), 0, sizeof({{STRUCT_NAME}}));";
     code_ += "  }";
 
-    // Generate a constructor that takes all fields as arguments.
+    // Generate a constructor that takes all fields as arguments,
+    // excluding arrays
     std::string arg_list;
     std::string init_list;
     padding_id = 0;
+    auto first = struct_def.fields.vec.begin();
     for (auto it = struct_def.fields.vec.begin();
          it != struct_def.fields.vec.end(); ++it) {
       const auto &field = **it;
+      if (IsArray(field.value.type)) {
+        first++;
+        continue;
+      }
+
```

Comment:
This PR introduces a lot of blank lines.. prefer to not have them by default, only if a method is really long. And even then, comments are probably a better separator.

Reply:
Removed most blank lines and added some comments.